adaptlearning/adapt-cli
# Rewrite and added AAT API

fixes #148 

### Added
* ESLint standard
* ~~Install from zip~~ note: should be removed due to compatibility issues with npm
* Install from folder
* API for AAT

### Changed
* lodash for lodash-es to reduce load time
* Converted to es6 import/export, var to let const
* Moved all imports to the top
* Reduced nesting
  * Used async/await instead of Q and promises throughout
  * Removed unnecessary wrappers and factories for hard linked files 
* Update command mimics install command

### Removed
* `npm` and `q` libraries

### Updated
* All dependencies



### Questions
* @taylortom As we're switching to npm at some point, I think I should probably remove zip installation as the github zip format isn't installable using npm, because of its nested folder i.e. package.json should be at the root.

### Testing
To test the AAT API:
```sh
git clone https://github.com/adaptlearning/adapt-cli
cd adapt-cli
git checkout develop
npm install
```
1. run this `$ node --inspect-brk ./lib/api.js --debug-wait`
2. go to `chrome://inspect` in your browser
3. wait for console log 'API Ready'
4. use `global.api` in the console to test the functions

To test the CLI:
1. `$ npm install -g adapt-cli@develop`
2. use as expected

To test the CLI from chrome devtools:
```sh
git clone https://github.com/adaptlearning/adapt-cli
cd adapt-cli
git checkout develop
npm install
```
1. run this `$ node --inspect-brk ./bin/adapt.js create course`
2. go to `chrome://inspect` in your browser
